# Data Exploration

In [ ]:
import pandas as pd

In [ ]:
# Glassdoor Companies Review Analysis
reviews = pd.read_csv('C:\Users\saite\OneDrive - University of Central Missouri\Documents\MS in CS\Data Analyst\Projects\Employee Rentention Analysis\Datasets\Glassdoor Reviews Cleaned.csv')

In [16]:
reviews.head()

Employeer ID Employeer Name Employement Status              job title  \
0       5224839  JPMorganChase            REGULAR  Private Client Banker   
1       5224839  JPMorganChase            REGULAR       Senior Associate   
2       5224839  JPMorganChase            REGULAR        Personal Banker   
3       5224839  JPMorganChase            REGULAR  Software Engineer III   
4       5224839  JPMorganChase            REGULAR                    NaN   

   length of employement         location  isCurrent Job  \
0                      9  Los Angeles, CA           True   
1                      6     New York, NY           True   
2                      4      Chicago, IL           True   
3                      2      Seattle, WA           True   
4                      1              NaN          False   

   ratingCarrerOpportunities ratingBusinessoutlook   ratingCEO  ...  \
0                          4              POSITIVE     APPROVE  ...   
1                          5              POSITIVE     APPROVE  ...   
2                          2                   NaN     APPROVE  ...   
3                          2              NEGATIVE  DISAPPROVE  ...   
4                          3                   NaN         NaN  ...   

   rating RecommendtoFriend  ratingSeniorleadership  rating worklifebalance  \
0                  POSITIVE                     4.0                     5.0   
1                  POSITIVE                     5.0                     5.0   
2                  NEGATIVE                     2.0                     3.0   
3                  NEGATIVE                     1.0                     4.0   
4                  NEGATIVE                     1.0                     2.0   

     reviewdatetime countHelpful  countNothelpful  \
0  18-05-2025 23:05            0                0   
1  31-05-2025 09:15            0                0   
2  23-05-2025 21:00            0                0   
3  13-05-2025 08:11            0                0   
4  23-05-2025 07:39            0                0   

                                             summary  \
0       Great Company, Location &amp; Team Dependent   
1                                 Good place to work   
2  Great Brand and Resources, But Support Varies ...   
3  Terrible developer experience in a low trust e...   
4                                      Disappointing   

                                              Advice  \
0  Increase base pay for private client bankers. ...   
1                                                NaN   
2  Advice to Management:\nInvest more in developi...   
3  Ditch the LLMs, focus on process efficiency, f...   
4  360 annual reviews for management \nExit inter...   

                                                Cons  \
0  Dependent on team and location. Lots of widget...   
1                                Pay is on lower end   
2  •\tThe employee experience depends heavily on ...   
3  Developer experience at this company is catast...   
4  Poor management who have been granted permissi...   

                                                pros  
0  Great place to learn the industry. If you're i...  
1       Friendly people, collaborative, good culture  
2  •\tThe company offers a well-known name and so...  
3                    Interesting problems to work on  
4  Compensation  \nIt looks good on your resume  ...  

[5 rows x 24 columns]

In [17]:
reviews.shape

(11925, 24)

# The dataset consists of 24 columns and 11925 Rows

In [18]:
reviews['Employeer Name'].nunique()

127

### By this we can conclude that we have the data from 127 different companies

In [19]:
reviews['Employeer Name'].value_counts()

Employeer Name
Elevance Health         109
NextEra Energy, Inc.    109
Centene                 109
ExxonMobil              100
Apple                   100
                       ... 
The Cigna Group          60
Alphabet (GB)            60
General Motors (GM)      60
Duke Energy              47
Applied Materials         9
Name: count, Length: 127, dtype: int64

### We extracted max of 110 reviews or less from each company 

In [20]:
companies = reviews['Employeer Name'].unique()
print(companies)

['JPMorganChase' 'Berkshire Hathaway' 'Amazon' 'Microsoft' 'Alphabet (GB)'
 'Apple' 'ExxonMobil' 'Wells Fargo' 'UnitedHealth Group' 'Walmart'
 'Goldman Sachs' 'Meta' 'Morgan Stanley' 'Citi' 'Verizon' 'AT&T' 'Comcast'
 'Johnson & Johnson' 'American Express' 'Tesla' 'Procter & Gamble'
 'CVS Health' 'General Motors (GM)' 'Broadcom' 'PepsiCo' 'The Home Depot'
 'Elevance Health' 'IBM' 'AbbVie' 'Cisco' 'The Cigna Group'
 'Costco Wholesale' 'The Coca-Cola Company' 'GE Aerospace' 'CATERPILLAR'
 'John Deere' 'Intel Corporation' 'ConocoPhillips' 'NVIDIA' 'RTX'
 'U.S. Bank' 'Ford Motor Company' 'Capital One' 'Apollo Global Management'
 'Visa Inc.' 'Thermo Fisher Scientific' 'Charles Schwab'
 'Progressive Insurance' 'Marathon Petroleum' 'The PNC Financial Services'
 'UPS' 'NextEra Energy, Inc.' 'Walt Disney Company' 'Phillips 66'
 'Metlife' 'Salesforce' 'Merck' 'Abbott' 'Dell Technologies' 'FedEx'
 'Lockheed Martin' 'Eli Lilly and Company' 'KKR'
 'Philip Morris International' 'Valero' 'BNY' 'Prude

### companies variable has the list of companies for which we extracted the reviews from glassdoor.
### Companies are the list of Forbes top 127 

# Now lets dive into the details of each column

In [21]:
reviews.columns

Index(['Employeer ID', 'Employeer Name', 'Employement Status', 'job title',
       'length of employement', 'location', 'isCurrent Job',
       'ratingCarrerOpportunities', 'ratingBusinessoutlook', 'ratingCEO',
       'ratingCompensationand Benefits', 'ratingCulutreand Values',
       'ratingDiversity and Inclusion', 'rating Overall',
       'rating RecommendtoFriend', 'ratingSeniorleadership',
       'rating worklifebalance', 'reviewdatetime', 'countHelpful',
       'countNothelpful', 'summary', 'Advice', 'Cons', 'pros'],
      dtype='object')

In [22]:
datatypes_reviews =reviews.dtypes

In [23]:
datatypes_reviews

Employeer ID                        int64
Employeer Name                     object
Employement Status                 object
job title                          object
length of employement               int64
location                           object
isCurrent Job                        bool
ratingCarrerOpportunities           int64
ratingBusinessoutlook              object
ratingCEO                          object
ratingCompensationand Benefits      int64
ratingCulutreand Values             int64
ratingDiversity and Inclusion     float64
rating Overall                    float64
rating RecommendtoFriend           object
ratingSeniorleadership            float64
rating worklifebalance            float64
reviewdatetime                     object
countHelpful                        int64
countNothelpful                     int64
summary                            object
Advice                             object
Cons                               object
pros                              